File: Testbed/LifeCycle/examples/counter.pl
#!/usr/bin/perl -w
use strict;
use JSON::XS;
use Getopt::Long;
use Clone qw(clone);

my ($in,$out,$json,$workflow,$payload);
$payload = {};
GetOptions(
                'in=s'  => \$in,
                'out=s' => \$out,
          );

open IN, "<$in" or die "open input $in: $!\n";
$json = <IN>;
close IN;
$payload = decode_json($json);
$workflow = $payload->{workflow};

$workflow->{counter}++;
print 'Counter: count=',$workflow->{counter},"\n";

open  OUT, ">$out" or die "open output $out: $!\n";
print OUT encode_json($payload);
close OUT;
exit 0;

File: Testbed/LifeCycle/examples/counter.py
#!/usr/bin/env python
import traceback, getopt, sys
import json, copy

try:
  opts, args = getopt.getopt(sys.argv[1:], "io", ["in=", "out="])
except getopt.GetoptError, err:
  print "Error parsing arguments"
  sys.exit(2)
_out = 'none'
_in  = 'none'
for o, a in opts:
  if o == "--out":
    _out = a
  elif o == "--in":
    _in = a
  else:
    assert False, "unhandled option"

f = open(_in,'r')
payload = json.loads(f.read())

try:
  payload['workflow']['counter'] += 1
except KeyError:
  payload['workflow']['counter'] = 1

print "Counter: count=",payload['workflow']['counter']

f = open(_out,'w')
f.write(json.dumps(payload))
